app/server/lib/SamlConfig.ts

```diff
@@ -204,6 +198,43 @@ export class SamlConfig {
     }));
   }
 
+  private async _processInitialRequest(req: express.Request) {
+    const relayState: string = req.body.RelayState;
+    const sessionId = this._gristServer.getSessions().getSessionIdFromRequest(req) || undefined;
+    const params = {
+      sessionId,
+      redirectUrl: "",
+      unsolicited: true,
+      action: "",
+    };
+
+    if (!relayState) {
+      // Presumably an IdP-inititated signin.
+      params.redirectUrl = checkRedirectUrl(relayState, req).href;
```

Comment:
This is making some computation with relayState after checking that relayState is falsy. Possibly unintentional?

Reply:
Yes, sorry, you're right, I accidentally swapped around these two redirect URLs.

Just fixed it.